src/datasources/network/network.module.ts

```diff
@@ -55,6 +55,27 @@ function fetchClientFactory(
       data,
     };
   };
+
+  /**
+   * A cache to prevent multiple in-flight requests for the same data.
+   */
+  const cache: Record<string, Promise<NetworkResponse<unknown>>> = {};
+
+  return async <T>(
+    url: string,
+    options: RequestInit,
+  ): Promise<NetworkResponse<T>> => {
+    // Naive key for simplicity but JSON.stringify is not stable
+    const key = JSON.stringify({ url, ...options });
```

Comment:
If the keys of an object are different in order, this key will be different, but I think this is a sufficient starting point.

Reply:
The key should be stable, according to your suggestion, as of 197e26611d8cb172cb8698dd2cd6b3c15271ab17.